infrastructure/src/main/java/org/corfudb/infrastructure/orchestrator/Action.java

```diff
@@ -0,0 +1,31 @@
+package org.corfudb.infrastructure.orchestrator;
+
+import org.corfudb.runtime.CorfuRuntime;
+
+import javax.annotation.Nonnull;
+import java.util.Map;
+
+/**
+ *
+ * A workflow action. All workflow actions must extend this class.
+ *
+ * Created by Maithem on 10/25/17.
+ */
+
+public interface Action {
+
+    /**
+     * Returns the name of this action.
+     * @return Name of action
+     */
+    @Nonnull
+    String getName();
+
+    /**
+     * Execute the action.
+     * @param runtime A runtime that the action will use to execute
+     * @return State that other dependent actions require to execute.
+     */
+    @Nonnull
+    void execute(@Nonnull CorfuRuntime runtime) throws Exception;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)